src/EntityFramework/Core/Objects/ELinq/ExpressionConverter.cs

```diff
@@ -1458,6 +1459,43 @@ private DbExpression ImplementEqualityUnknownArguments(DbExpression left, DbExpr
         #region Helper Methods Shared by Translators
 
         // <summary>
+        // Helper method for String.Like
+        // object.Like(likeExpression[, escapeCharacter]) is translated to:
+        // object like likeExpression [escape escapeCharacter]
+        // </summary>
+        // <returns> The translation </returns>
+        private DbExpression TranslateLike(MethodCallExpression call)
+        {
+            char dummyEscapeChar;
+            var providerSupportsEscapingLikeArgument = ProviderManifest.SupportsEscapingLikeArgument(out dummyEscapeChar);
+
+            var inputExpression = call.Arguments[0];
+            var patternExpression = call.Arguments[1];
+            var escapeExpression = (call.Arguments.Count > 2 ? call.Arguments[2] : null);
+            
+            if (!providerSupportsEscapingLikeArgument && (escapeExpression != null))
+            {
+                throw new ProviderIncompatibleException(Strings.ProviderDoesNotSupportEscapingLikeArgument);
+            }
+
+            var translatedPatternExpression = TranslateExpression(patternExpression);
+            var translatedEscapeExpression = (escapeExpression != null ? TranslateExpression(escapeExpression) : null);
+            var translatedInputExpression = TranslateExpression(inputExpression);
+
+            DbExpression result;
+            if (escapeExpression != null)
```

Comment:
convert this to:
return escapeExpression != null
? translatedInputExpression.Like(translatedPatternExpression, translatedEscapeExpression)
: translatedInputExpression.Like(translatedPatternExpression);